clippy_lints/src/needless_pass_by_value.rs

```diff
@@ -132,27 +132,25 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for NeedlessPassByValue {
                         db.span_suggestion(input.span,
                                         &format!("consider changing the type to `{}`", slice_ty),
                                         slice_ty);
-                        return;
+                        return; // `Vec` and `String` cannot be destructured - no need for `*` suggestion
```

Comment:
Please add  assertions that `span_need_deref` is empty as a canary for when matching on vec/string becomes possible.